doc/spec/manifest.md

```diff
@@ -0,0 +1,245 @@
+    
+        A URL to a maintainer's website, blog, etc.
+
+- **`description`** *string*
+
+    A brief description of the content or application.
+
+- **`keywords`** *array*
+
+    An array of strings which should be considered keywords that may be used
+    by a separate search system to index content.
+
+- **`website`** *string*
+
+    A URL to a related website or project page for the application.
+
+- **`licenses`** *array*
+
+    An array of strings which identify any software licenses which apply
+    to the application. Values should be a [SPDX License Identifier](https://spdx.org/licenses/).
+
+- **`Dependencies`** *array*
+
+    An array of objects which describe content which this application depends
+    on. Those may be generic blobs of data or other content manifests and
+    their dependencies as described by their media type. Fields are:
+    
+    - **`mediaType`** *string*
+    
+        The MIME type of the dependency object.
+    
+    - **`size`** *int*
+    
+        The size in bytes of the dependency object.
+    
+    - **`repository`** *string*
+    
+        The name of the repository from which to fetch the object.
+
+    - **`encryption`** *object*
+
+        An object containing information about the algorithm, key, and
+        initialization vector used to decrypt the content. If omitted,
+        the content should be assumed to be unencrypted. Fields are:
+
+        - **`algorithm`** *string*
+
+            Identifies the  encryption algorithm used.
+
+        - **`key`** *string*
+
+            The hex encoding of the key required to decrypt the content.
+
+        - **`iv`** *string*
+
+            The hex encoding of the initialization vector required to decrypt
+            the content. If omitted and an initialization vector is required
+            for the specified algorithm, assume an all-zero initialization
+            vector; this may be used to support convergent encryption.
+    
+    - **`digest`** *string*
+    
+        A string labeled with a digest algorithm prefix and the hex-encoded
+        value of that digest for the content. If the content is encrypted,
+        then the digest is of the encrypted content.
+
+        Digest strings should be of the form:
+
+            <digest-alg-name>:<hex-encoded-digest>
+    
+    Any media type may be in the `mediaType` field as the application may
+    use any type of content. If the `mediaType` is
+    `application/vnd.docker.distribution.content.manifest.v2+json` then the
+    dependency is another content manifest as described by this document. The
+    dependencies from that manifest should also be considered dependencies of
+    this application.
+    
+    If the `repository` field is empty or omitted, then the object should be
+    considered to be in the same repository as this manifest.
+
+    Encryption combined with a verifiable digest of the content should allow
+    for both confidentiality and authentication of data.
```

Comment:
We should probably just mandate [authenticated encryption!](http://en.wikipedia.org/wiki/Authenticated_encryption) modes so that we get authentication as part of the content decryption.